src/kernel/src/process/file.rs

```diff
@@ -68,4 +72,22 @@ impl FileDesc {
 
         files.get(fd)?.clone()
     }
+
+    pub fn free(&self, fd: Fd) -> Result<(), SysErr> {
+        if fd.get() < 0 {
+            return Err(SysErr::Raw(EBADF));
+        }
```

Comment:
This check should be a responsibility of `Fd` type like `str` in Rust that it will never allow non-UTF8.

Reply:
You mean that the check should be in Fd::new?

Reply:
If so, should this function receive an Fd, or an i32 with Fd::new being called inside the function?

Reply:
Better to accept an `Fd` so the caller know it need to pass a file descriptor.

Reply:
Couldn't that diverge too much from what the actual kernel does?

Reply:
I agree, it's better to not have the wrapper in the first place.